From ba9bca4765be8b15633f6b87e75fbc8d5b27726b Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 26 Aug 2019 16:12:05 -0400
Subject: [PATCH] console: Fix a typo in the EFI warning list in gnu-efi

Some versions of gnu-efi have a typo, in which "EFI_WARN_UNKNOWN_GLYPH"
is accidentally "EFI_WARN_UNKOWN_GLYPH".  Work around that, so that we
can use the not-silly one in console.c's list of error and warning
messages.

This is a backport from devel for:

  commit 5f62b22ccd636d326b3229a2b196118701c6f3f7
  Author: Peter Jones <pjones@redhat.com>
  Date:   Mon Aug 26 16:12:05 2019 -0400

Signed-off-by: Peter Jones <pjones@redhat.com>

Upstream-Status: Backport
[https://github.com/rhboot/shim/commit/d230d02f990f02293736dca78b108f86c86d1bd0]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 include/console.h | 6 ++++++
 lib/console.c     | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/console.h b/include/console.h
index ac6fdf61..137fcd3d 100644
--- a/include/console.h
+++ b/include/console.h
@@ -7,6 +7,12 @@
 #define PrintAt(fmt, ...) \
 	({"Do not directly call PrintAt() use console_print_at() instead" = 1;});
 
+#if !defined(EFI_WARN_UNKNOWN_GLYPH) && defined(EFI_WARN_UNKOWN_GLYPH)
+#define EFI_WARN_UNKNOWN_GLYPH EFI_WARN_UNKOWN_GLYPH
+#elif !defined(EFI_WARN_UNKNOWN_GLYPH)
+#define EFI_WARN_UNKNOWN_GLYPH EFIWARN(1)
+#endif
+
 EFI_STATUS
 console_get_keystroke(EFI_INPUT_KEY *key);
 UINTN
diff --git a/lib/console.c b/lib/console.c
index c92d27f3..9e92e76e 100644
--- a/lib/console.c
+++ b/lib/console.c
@@ -604,7 +604,7 @@ static struct {
 	{  EFI_SECURITY_VIOLATION,     L"Security Violation"},
 
 	// warnings
-	{  EFI_WARN_UNKOWN_GLYPH,      L"Warning Unknown Glyph"},
+	{  EFI_WARN_UNKNOWN_GLYPH,     L"Warning Unknown Glyph"},
 	{  EFI_WARN_DELETE_FAILURE,    L"Warning Delete Failure"},
 	{  EFI_WARN_WRITE_FAILURE,     L"Warning Write Failure"},
 	{  EFI_WARN_BUFFER_TOO_SMALL,  L"Warning Buffer Too Small"},
-- 
2.25.1

